web/source/dom/targets/contentEditable.ts

```diff
@@ -107,7 +107,7 @@ namespace com.keyman.dom.targets {
 
     getTextBeforeCaret(): string {
       if(!this.hasSelection()) {
-        return;
+        return this.getText();
```

Comment:
This matches the fix seen here, which is already part of 17.0:

https://github.com/keymanapp/keyman/blob/66142be3930c24f996939f7496a377f23223e3c3/web/src/engine/element-wrappers/src/contentEditable.ts#L109-L112